src/graph.rs

```diff
@@ -231,24 +255,24 @@ mod tests {
         // Regression test for https://github.com/bertptrs/tracing-mutex/issues/7
         let mut graph = DiGraph::default();
 
-        assert!(!graph.add_edge(1, 1));
+        assert_eq!(graph.add_edge(1, 1, ()), Err(vec![(1, ())]));
```

Comment:
The tests do not compile since you later changed the `add_edge` signature to take an `FnOnce` rather than a value.